src/firewall/core/nftables.py

```diff
@@ -399,13 +399,14 @@ def flush_rules(self, log_denied):
         self.rich_rule_priority_counts = {}
         self.policy_dispatch_index_cache = {}
 
-        rules = []
-        if TABLE_NAME in self.created_tables["inet"]:
-            rules.append({"delete": {"table": {"family": "inet",
-                                               "name": TABLE_NAME}}})
-            self.created_tables["inet"].remove(TABLE_NAME)
-
-        self.set_rules(rules, log_denied=log_denied)
+        try:
+            self.set_rule({"delete": {"table": {"family": "inet", "name": TABLE_NAME}}},
+                          log_denied=log_denied)
+        except ValueError:
```

Comment:
Please split the patch.

Most of the patch is about replacing `self.created_tables` with `self.policy_table_active`.
As such, here we should do:

```python
if self.policy_table_active:
    self.policy_table_active = False
    self.set_rule(...)
```

The change with `ValueError` may still be good, but please do that in a follow-up patch, with a proper commit message.

--- 

In particular, the patch is relatively large but has a `Fixes: rhbz2222044` comment. As most of the change is an internal refactoring, the relevant part (of always calling `self.set_rule()`) should be in a separate/smaller patch.

Reply:
also, before (after?) deleting the table, we should always set `self.policy_table_active = False`

Reply:
I will attempt to split it, although I think you're misunderstanding the original bug. This is not just a refactor. The bug is _caused_ by `created_tables`. Anyways. I think I can simplify it a bit.